applications/GeoMechanicsApplication/custom_constitutive/thermal_dispersion_2D_law.cpp

```diff
@@ -0,0 +1,67 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#include "custom_constitutive/thermal_dispersion_2D_law.hpp"
+#include "geo_mechanics_application_variables.h"
+#include <iostream>
+
+namespace Kratos {
+
+ConstitutiveLaw::Pointer GeoThermalDispersion2DLaw::Clone() const
+{
+    return Kratos::make_shared<GeoThermalDispersion2DLaw>(*this);
+}
+
+SizeType GeoThermalDispersion2DLaw::WorkingSpaceDimension()
+{
+    constexpr SizeType space_dimension = 2;
```

Comment:
For the space_dimension, I would propose to have a member variable in the class, mNumberOfDimensions that we set in the constructor via arguments to either 2 or 3, such that this class knows its own dimensionality.